packages/ember-glimmer/lib/environment.js

```diff
@@ -256,6 +258,14 @@ export default class Environment extends GlimmerEnvironment {
       assert(`A helper named "${key}" could not be found`, !isBlock || this.hasHelper(key, parentMeta));
     }
 
+    if ((!isSimple && appendType === 'unknown') || appendType === 'self-get') {
+      return statement.original.deopt();
+    }
+
+    if (!isSimple && path) {
+      return DynamicComponentSyntax.fromPathReference({ path, args, templates, parentMeta });
```

Comment:
pathReference is a bit confusing, would call it `fromPath`, `fromPathAndInvocationArgs`, ... (just not "path reference" because that refers to another thing elsewhere